File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.py
class Solution:
    def firstocc(self, arr: List[int], size: int, n: int):
        low=0
        high=size-1
        first=-1
        while low<=high:
            mid= low+(high-low)//2
            if arr[mid]==n:
                first= mid
                high= mid-1
            elif arr[mid]<n:
                low= mid+1
            else:
                high= mid-1
        return first
    def lastocc(self, arr: List[int], size: int, n: int):
        low=0
        high= size-1
        last=-1
        while low<=high:
            mid= low+(high-low)//2
            if arr[mid]==n:
                last= mid
                low= mid+1
            elif arr[mid]<n:
                low= mid+1
            else:
                high= mid-1
        return last
    def searchRange(self, nums: List[int], target: int) -> List[int]:
        if self.firstocc(nums, len(nums), target)==-1:
            return [-1,-1]
        return [self.firstocc(nums, len(nums), target), self.lastocc(nums, len(nums), target)]

File: 0034-find-first-and-last-position-of-element-in-sorted-array/0034-find-first-and-last-position-of-element-in-sorted-array.cpp
class Solution {
private:
    int firstocc(vector<int>& arr, int size, int n){
        int low=0;
        int high=size-1;
        int first=-1;
        while (low<=high){
            int mid= low+((high-low)/2);
            if (arr[mid]==n){
                first= mid;
                high= mid-1;
            }
            else if (arr[mid]<n){
                low= mid+1;
            }
            else{
                high= mid-1;
            }
        }
        return first;
    }
    int lastocc(vector<int>& arr, int size, int n){
        int low=0;
        int high=size-1;
        int last= -1;
        while (low<=high){
            int mid= low+((high-low)/2);
            if (arr[mid]==n){
                last= mid;
                low= mid+1;
            }
            else if (arr[mid]<n){
                low= mid+1;
            }
            else{
                high= mid-1;
            }
        }
        return last;
    }
public:
    vector<int> searchRange(vector<int>& nums, int target) {
        if (firstocc(nums, nums.size(), target)==-1){
            return {-1,-1};
        }
        return {firstocc(nums, nums.size(), target), lastocc(nums, nums.size(), target)};
    }
};